corehq/ex-submodules/dimagi/utils/couch/migration.py

```diff
@@ -126,7 +128,9 @@ def save(self, *args, **kwargs):
         if sync_to_sql:
             try:
                 self._migration_do_sync()
-            except:
+            except Exception as e:
+                if settings.UNIT_TESTING:
+                    raise e
                 sql_class_name = self._migration_get_sql_model_class().__name__
                 couch_class_name = self.__class__.__name__
                 notify_exception(None,
```

Comment:
Another way to do this would be to have tests that need this to raise an error patch `notify_exception()` with a function that re-raises the error. That would be cleaner because then this code would not need to reference `settings.UNIT_TESTING`.

Reply:
I'm inclined to leave this alone - I'd rather have this fail loudly as often as possible, even if the code is a little bit messier. It's easy to decorate individual tests now, when there are few relevant tests, but when we migrate models that are used more broadly, this may fail in tests that aren't the tests written specifically for the migration.